### DISCUSSION TITLE
jdownloader cask formula installs incompatible application on Apple Silicon (M2)

### DISCUSSION THREAD

**Discussion by bjtn1** [Homebrew/discussions#5265] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.16
ORIGIN: https://github.com/Homebrew/brew
HEAD: bd1155be8f50998429a795c15a69c8fe75250510
Last commit: 4 days ago
Core tap JSON: 05 Apr 22:38 UTC
Core cask tap JSON: 05 Apr 22:38 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.44.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

The formula for JDownloader on Apple Silicon doesn't seem right.

Running `brew install --cask jdownloader` followed by opening the recently installed JDownloader cask results in the following error message:

<img width="1470" alt="image" src="https://github.com/Homebrew/discussions/assets/58859142/3f2ffc7c-a3c1-42a0-8f80-4ab1bcca43a3">

Additionally, [the Homebrew formula for jdownloader](https://github.com/Homebrew/homebrew-cask/blob/7159db0a23e6ac3204015917f256569a8b575110/Casks/j/jdownloader.rb) seems to get the `.dmg` file from a site different than [JDownloader's official download site](https://jdownloader.org/download/index).

Accessing the url found in the Homebrew formula for this cask downloads a `JDSetup.dmg` file that contains a JDownloader installer that fails to install the cask on Apple Silicon with the following error messages:

<img width="1327" alt="image" src="https://github.com/Homebrew/discussions/assets/58859142/c802ce6d-4bf0-4d44-9bb3-385a96411b7d">

The `.log` file contains the following error (System properties have been cut off for privacy reasons)

<img width="1470" alt="image" src="https://github.com/Homebrew/discussions/assets/58859142/131df6e3-6958-4357-81ee-e465cd1dcbc7">

Lastly, [JDownloader's official download site](https://jdownloader.org/download/index) requires Apple Silicon users to download the file from `mega.nz`, which differs from the `url` field in the current Homebrew formula for jdownloader.

The official JDownloader installer and app work as intended on Apple Silicon.

Perhaps I'm doing something very wrong when installing JDownloader through Homebrew. Has this happened to anyone else? Does the Homebrew formula need to be updated? Thank you for your time.

### Relevant casks

jdownloader

**Comment by daeho-ro** [comment, marked as answer]
It is already reported and opened a PR for that.
https://github.com/Homebrew/homebrew-cask/issues/163482

**Reply by bjtn1** [reply]
Thank you!
